### PR TITLE
#10: Added a module for code coverage in UDFs

### LINKED ISSUE #10: Add code-coverage support

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +34,14 @@ System property: `test.debug`
 
 This module instructs the UDF JVMs to connect to a Java debugger listening on the default port 8000 on you machine, running the tests.
 
+### Code Coverage
+
+System property: `test.coverage`
+
+This module installs a jacoco agent to the UDF JVM and receives the execution data using a TCP socket.
+
+This module requires some maven configuration. Use the [project-keeper's](https://github.com/exasol/project-keeper-maven-plugin) `udf_coverage` module to verify it.
```
Comment (morazow) [inline README.md:43]:
```suggestion
This module requires additional maven configuration. Use the [project-keeper's](https://github.com/exasol/project-keeper-maven-plugin) `udf_coverage` module to verify it.   
```

File: doc/changes/changes_0.3.0.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# udf-debugging-java 0.3.0, released 2020-11-20
```
Comment (morazow) [inline doc/changes/changes_0.3.0.md:1]: Please do not forget to change the date.

File: doc/changes/changes_0.3.0.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ 
+Code name: Test setup for UDF code coverage
+
+## Features / Enhancements
+
+* #10: Added a module for code coverage in UDFs 
+
+## Dependency Updates:
+
+* Updated `com:exasol:project-keeper-maven-plugin`:0.2.0 to 0.3.0
```
Comment (morazow) [inline doc/changes/changes_0.3.0.md:11]: From the internal discussion, I think we agreed on this format:
```
Updated to `com:exasol:project-keeper-maven-plugin:0.3.0` (was 0.2.0)
```

Comment (morazow) [reply doc/changes/changes_0.3.0.md:11]: Please feel free to change it.

File: src/main/java/com/exasol/udfdebugging/modules/coverage/CoverageModule.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        return Stream.of(this.jvmOption);
+    }
+
+    private void uploadAgentToBucketFs(final Bucket bucket) {
+        try {
+            bucket.uploadFile(JACOCO_AGENT_PATH, JACOCO_AGENT_NAME);
+        } catch (final TimeoutException | BucketAccessException exception) {
+            throw getUploadFailedException(exception);
+        } catch (final InterruptedException exception) {
+            Thread.currentThread().interrupt();
```
Comment (morazow) [inline src/main/java/com/exasol/udfdebugging/modules/coverage/CoverageModule.java:55]: :+1:

**Review by morazow — CHANGES_REQUESTED** [review]

**Review by morazow — APPROVED** [review]
Approved. 
There seems to be some Sonar issues.

**Comment by jakobbraun** [comment]
only coverage.

**Pull request merged**